src/client/Ruby/ms-rest/lib/ms_rest/http_operation_request.rb

```diff
@@ -68,23 +68,19 @@ def initialize(base_uri, path_template, method, options = {})
     # @return [URI] body the HTTP response body.
     def run_promise(&block)
       Concurrent::Promise.new do
-        @connection ||= Faraday.new(:url => base_uri, :ssl => MsRest.ssl_options) do |faraday|
-          middlewares.each{ |args| faraday.use(*args) } unless middlewares.nil?
-          faraday.adapter Faraday.default_adapter
-          logging = ENV['AZURE_HTTP_LOGGING'] || log
-          if logging
-            faraday.response :logger, nil, { :bodies => logging == 'full' }
-          end
-        end
+        options = {
+            :middlewares => middlewares,
+            :log => log
```

Comment:
Faraday supports log options (https://github.com/lostisland/faraday/blob/master/lib/faraday/response/logger.rb), HttpOperationRequest creation allows for `:log` to be passed as part of the options. If set, then we should pass that when creating a Faraday connection.